kube-client/src/config/file_config.rs

```diff
@@ -328,6 +329,24 @@ impl Kubeconfig {
     }
 }
 
+impl FromStr for Kubeconfig {
```

Comment:
This feels a bit redundant with `from_text` (especially given that both have essentially meaningless names). I'm also not really sure I'm comfortable blessing this as "the" supported format. Even if YAML is pretty much the canonical format I'd rather have this be something that the user has to acknowledge explicitly.

Reply:
This helps when used via `str::parse`. It also could be made intelligent enough to try multiple formats and use the one that works.
However, if the consensus is that this is controversial - I can drop it altogether. LMK

Reply:
I agree with Teo here as well. If we have an explicit method on `Kubeconfig::from_yaml_str` (i believe the latest suggestion is) then the user knows what tradeoffs they are making, and what format the string is supposed to be.

Reply:
Very well, I'll drop it